corehq/apps/hqwebapp/views.py

```diff
@@ -1348,3 +1351,62 @@ def log_email_event(request, secret):
     handle_email_sns_event(message)
 
     return HttpResponse()
+
+
+@method_decorator(require_superuser, name="dispatch")
+class OauthApplicationRegistration(BasePageView):
+    urlname = 'oauth_application_registration'
+    page_title = "Oauth Application Registration"
+    template_name = "hqwebapp/oauth_application_registration_form.html"
+
+    @property
+    def page_url(self):
+        return reverse(self.urlname)
+
+    @property
+    def base_application_form(self):
+        return modelform_factory(
+            get_application_model(),
+            fields=(
+                "name",
+                "client_id",
+                "client_secret",
+                "client_type",
+                "authorization_grant_type",
+                "redirect_uris",
+            ),
+        )
+
+    @property
+    def hq_application_form(self):
```

Comment:
These two properties could be static. If they were, could they be used to create crispy forms? ... And would crispy forms be better than the HTML in `oauth_application_registration_form.html`?

These are genuine questions, because maybe crispy forms are not the right approach here.

Reply:
Crispy forms may have been the better way of going about it, although I always find them to be more opaque and frustrating to work with than just writing out the HTML, even if the crispy way is DRYer.

Reply:
Happy to revisit it if you feel strongly though.